Data/Conduit/Async.hs

```diff
@@ -74,3 +79,78 @@ drainTo size gather = do
         case mx of
             Just x  -> return x
             Nothing -> scatter worker chan
+
+-- | Concurrently process input by spawning worker threads.
```

Comment:
Is it possible to describe semantics of and guarantees of the `mapConcurrently` otherwise it's not possible to understand them without reading the code?

Reply:
I just added what the implementation tries to do.